elmslie-core/src/main/java/vivid/money/elmslie/core/store/binding/ConversionContract.kt

```diff
@@ -15,60 +14,62 @@ class ConversionContract<InitiatorEvent, InitiatorEffect, InitiatorState,
 ) {
 
     private val disposable = CompositeDisposable()
-    private val contracts = mutableSetOf<() -> Unit>()
+    private val contracts = mutableSetOf<() -> Disposable>()
 
     /**
-     * Defines full state conversion between stores
+     * Defines full direct state conversion between stores.
      */
     fun states(
         conversion: InitiatorState.() -> ResponderEvent? = { null }
     ) = states({ this }, conversion)
 
     /**
-     * Defines full effect conversion between stores
+     * Defines full direct effect conversion between stores.
      */
     fun effects(
         conversion: InitiatorEffect.() -> ResponderEvent? = { null }
     ) = effects({ this }, conversion)
 
     /**
-     * Defines encrypted state conversion between stores
+     * Defines partial encrypted state conversion between stores.
      */
     fun <EncryptedState> states(
-        cypher: Observable<InitiatorState>.() -> Observable<EncryptedState>,
+        cypher: InitiatorState.() -> EncryptedState?,
         conversion: EncryptedState.() -> ResponderEvent? = { null }
-    ) {
-        initiator.states.cypher() to responder with conversion
-    }
+    ) = contract(initiator::states, cypher, conversion)
 
     /**
-     * Defines encrypted effect conversion between stores
+     * Defines partial encrypted effect conversion between stores.
      */
     fun <EncryptedEffect> effects(
-        cypher: Observable<InitiatorEffect>.() -> Observable<EncryptedEffect>,
+        cypher: InitiatorEffect.() -> EncryptedEffect?,
```

Comment:
This an incubating feature, because there are vary few usage examples. Essentially, this allows to map or transform data passed between stores. Naming may change in future versions.

Reply:
I suggest to discuss this when there will be more cases for changind data passed between stores to find a suitable naming convention.